### PR TITLE
Replace registration.sh with go-based side-car

### PR DESCRIPTION
**Pull request by lobkovilya**
## Motivation and Context

Go side-car is more convenient way to register SPIFFE entries. Also side-car will be useful for configuring spire-servers across several clouds. 
## How Has This Been Tested?

- [ ] Covered by existing integration testing
- [ ] Added integration testing to cover
- [X] Tested locally
- [ ] Have not tested

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: side-cars/cmd/nsm-spire/main.go [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+	var err error
+	c := tools.NewOSSignalChannel()
+
+	var serverConn *grpc.ClientConn
+	for {
+		serverConn, err = tools.DialUnixInsecure(serverSocketPath)
+		if err == nil {
+			break
+		}
+		logrus.Errorf("failed to dial server: %v", err)
```
Comment (haiodo) [inline side-cars/cmd/nsm-spire/main.go]: I think a short delay should be good idea to be added here.

**Comment by nickolaev** [comment]
I am afraid we are proliferating side-cars too much. Shall we have a unified one?

File: side-cars/cmd/nsm-spire/main.go (hunk 1 above)
Comment (lobkovilya) [reply side-cars/cmd/nsm-spire/main.go]: `tools.DialUnixInsecure` is blocking with timeout 15 sec by default

**Comment by lobkovilya** [comment]
@nickolaev that's interesting point, what would you like to see in unified side-car?

**Comment by haiodo** [comment]
I think yes we could have unified one, for monitor/spire/dns stuff all together.

**Pull request merged**